### DISCUSSION TITLE
Shader Pipeline Disabled

### DISCUSSION THREAD

**Discussion by bvv4ai** [openscenegraph/OpenSceneGraph#1224] [Q&A]
Hello All! 
I AM new to OpenSceneGraph. I am trying to view a scenegraph using the osgviewer and I am getting the following output on the terminal.

void StateSet::setGlobalDefaults() ShaderPipeline disabled.
void StateSet::setGlobalDefaults() ShaderPipeline disabled.

The image rendered is not rendered. But the same image is rendered if I use do it using a program.

**Comment by robertosfield** [comment]
There is experimental shader composition code in OpenSceneGraph master that reports the debug message.  Please use the last stable release as a base for testing. i.e OSG-3.6.5.

Also consider the VulkanSceneGraph, iit's the successor to the OpenSceneGraph is much better placed for new software projects. The OpenSceneGraph role is now supporting existing applications.
